### PR TITLE
Abstract Ops is added

### PR DESCRIPTION
**Pull request by nishith-vihar**
Abstract Ops is added and ops libraries are accodingly modified
to inherit the abstract ops.

Fixes: #153
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

### CODE REVIEW

**Comment by srijan-sivakumar** [comment]
@nishith-vihar , let's take this to draft stage till we take it up again.

**Comment by nishith-vihar** [comment]
> @nishith-vihar , let's take this to draft stage till we take it up again.

Ack

**Comment by srijan-sivakumar** [comment]
As the logging now doesn't go into the file level details, @nishith-vihar , I think we can pursue this PR..

**Comment by srijan-sivakumar** [comment]
@nishith-vihar , try to rebase and see what all new changes went through..

File: tests/functional/glusterd/test_peer_probe_detach.py [hunk 1]
```diff
@@ -25,5 +25,5 @@ def run_test(self, redant):
         server3 = self.server_list[2]
         redant.create_cluster(self.server_list)
         redant.pool_list(server1)
-        redant.peer_detach(server1, server2)
-        redant.peer_detach(server3, server2)
+        redant.peer_detach(server2,server1)
```
Comment (srijan-sivakumar) [inline tests/functional/glusterd/test_peer_probe_detach.py:28]: We actually have a destroy cluster module in the framework, I suggest we use that here

File: support/ops/gluster_ops/peer_ops.py [hunk 2]
```diff
@@ -30,16 +31,9 @@ def peer_probe(self, server: str, node: str):
...
-        ret = self.execute_command(cmd, node)
-
-        if ret['error_code'] != 0:
-            self.logger.error(ret['msg']['opErrstr'])
-            raise Exception(ret['msg']['opErrstr'])
-
-        self.logger.info(f"Successfully ran {cmd} on {node}")
-
+        
+        ret = self.execute_abstract_op_node(cmd,node)
```
Comment (srijan-sivakumar) [inline support/ops/gluster_ops/peer_ops.py:35]: nit: ret = self.execute_abstract_op_node(cmd, node)

File: support/ops/gluster_ops/volume_ops.py [hunk 3]
```diff
@@ -36,19 +36,11 @@ def volume_mount(self, server: str, volname: str,
...
-
-        ret = self.execute_command(cmd, node)
-
-        if int(ret["error_code"]) != 0:
-            self.logger.error(ret["error_msg"])
-            raise Exception(ret["error_msg"])
-
-        self.logger.info(f"Successfully ran {cmd} on {node}")
-
+        ret = self.execute_abstract_op_node(cmd,node)
```
Comment (srijan-sivakumar) [inline support/ops/gluster_ops/volume_ops.py:39]: NIT: add a space after `,`

**Review by srijan-sivakumar — CHANGES_REQUESTED** [review]
Do check the added comments.

File: support/ops/gluster_ops/peer_ops.py (hunk 2 above)
Comment (nishith-vihar) [reply support/ops/gluster_ops/peer_ops.py:35]: Ack

File: support/ops/gluster_ops/volume_ops.py (hunk 3 above)
Comment (nishith-vihar) [reply support/ops/gluster_ops/volume_ops.py:39]: Ack

File: tests/functional/glusterd/test_peer_probe_detach.py (hunk 1 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_peer_probe_detach.py:28]: Ack

**Review by srijan-sivakumar — APPROVED** [review]

**Pull request merged**